common/arg.cpp

```diff
@@ -3423,5 +3423,36 @@ common_params_context common_params_parser_init(common_params & params, llama_ex
         }
     ).set_examples({LLAMA_EXAMPLE_SERVER}));
 
+    // diffusion parameters
+    add_opt(common_arg({ "--diffusion-steps" }, "N",
+                       string_format("number of diffusion steps (default: %d)", params.diffusion.steps),
+                       [](common_params & params, int value) { params.diffusion.steps = value; })
+                .set_examples({ LLAMA_EXAMPLE_DIFFUSION }));
+    add_opt(
+        common_arg({ "--diffusion-eps" }, "F",
+                   string_format("epsilon for timesteps (default: %.6f)", (double) params.diffusion.eps),
+                   [](common_params & params, const std::string & value) { params.diffusion.eps = std::stof(value); })
+            .set_examples({ LLAMA_EXAMPLE_DIFFUSION }));
+    add_opt(common_arg(
+                { "--diffusion-algorithm" }, "N",
+                string_format("diffusion algorithm: 0=ORIGIN, 1=MASKGIT_PLUS, 2=TOPK_MARGIN, 3=ENTROPY (default: %d)",
+                              params.diffusion.algorithm),
+                [](common_params & params, int value) { params.diffusion.algorithm = value; })
+                .set_examples({ LLAMA_EXAMPLE_DIFFUSION }));
+    add_opt(common_arg(
+                { "--diffusion-alg-temp" }, "F",
+                string_format("algorithm temperature (default: %.3f)", (double) params.diffusion.alg_temp),
+                [](common_params & params, const std::string & value) { params.diffusion.alg_temp = std::stof(value); })
+                .set_examples({ LLAMA_EXAMPLE_DIFFUSION }));
+    add_opt(common_arg({ "--diffusion-max-length" }, "N",
+                       string_format("maximum generation length (default: %d)", params.diffusion.max_length),
+                       [](common_params & params, int value) { params.diffusion.max_length = value; })
+                .set_examples({ LLAMA_EXAMPLE_DIFFUSION }));
```

Comment:
This argument is not needed - use `n_ubatch` to control the max diffusion length.